benchmarks/src/main/java/io/druid/benchmark/package-info.java

```diff
@@ -17,17 +17,7 @@
  * under the License.
  */
 
-package io.druid.segment;
+@EverythingIsNonnullByDefault
+package io.druid.benchmark;
 
-import io.druid.guice.annotations.ExtensionPoint;
-import io.druid.segment.serde.Serializer;
-
-import java.io.IOException;
-
-@ExtensionPoint
-public interface GenericColumnSerializer extends Serializer
```

Comment:
This was marked as an extension point since it's a return value of ComplexMetricSerde's `getSerializer` method. So I think we should revert this change. I also am not sure it's worth making even in the next major release, since there isn't a huge benefit and it would mean people need to adjust their extensions.

Reply:
Next release is a major release (0.12.x -> 0.13). I don't think we should fall into conservatism (not fixing things because users may depend on them) until Druid 1.0, and clean up APIs eagerly until before that.

Reply:
IMO it is painful for users to have to make small adjustments for non-functionality-affecting changes each time they upgrade to a new major Druid release. So I think it would be friendlier to do a 'big bang' and make any desired non-functionality-affecting, yet still backwards incompatible, changes like this all in one major release. Probably the one immediately prior to 1.0.

Reply:
Ok. #6103.